File: notebooksjava/_16BitFlipMaxConsecutiveOnes.java
public class _16BitFlipMaxConsecutiveOnes {
    public int reverseBits(int num) {
        // 🏁 特权拦截哨兵：如果 32 位全满载（即 -1）
        if (num == -1) return 32;
        
        int current = 0;
        int flipped = 0;
        int maxLen = 0;
        
        // 🚂 严格走满 32 位有符号车厢的生命周期
        for (int i = 0; i < 32; i++) {
            // 🔬 显微镜：检查最右边的一格是不是 1
            if ((num & 1) == 1) {
                current++;
                flipped++;
            } else {
                // 🌪️ 大盘暴跌与逻辑交接时刻
                flipped = current + 1;
                current = 0;
            }
            
            // 🛡️ 档案馆：实时锁死历史最高纪录
            maxLen = Math.max(maxLen, flipped);
            
            // 🚨 【高能核心改动】：必须用无符号右移 >>> 
            num >>>= 1;
        }
        
        return maxLen;
    }

    // 🚦 工业级自动化测试集
    public static void main(String[] args) {
        _16BitFlipMaxConsecutiveOnes sol = new _16BitFlipMaxConsecutiveOnes();
        System.out.println("🚦 开始执行 Java 位运算算法硬核全测...");

        // 用 Java 的 assert（运行时需加上 -ea 参数）
        // 或者直接打印验证结果是否符合我们讨论的物理真理
        
        System.out.println("Case 1 (1775): " + (sol.reverseBits(1775) == 8 ? "✅ 通过" : "❌ 失败"));
        System.out.println("Case 2 (7):    " + (sol.reverseBits(7) == 4 ? "✅ 通过" : "❌ 失败"));
        System.out.println("Case 3 (-1):   " + (sol.reverseBits(-1) == 32 ? "✅ 通过" : "❌ 失败"));
        System.out.println("Case 4 (-2):   " + (sol.reverseBits(-2) == 32 ? "✅ 通过" : "❌ 失败"));
        System.out.println("Case 5 (-3):   " + (sol.reverseBits(-3) == 32 ? "✅ 通过" : "❌ 失败（我们算对了，32才是真理！）"));
        System.out.println("Case 6 (-8):   " + (sol.reverseBits(-8) == 30 ? "✅ 通过" : "❌ 失败"));
        System.out.println("Case 7 (0):    " + (sol.reverseBits(0) == 1 ? "✅ 通过" : "❌ 失败"));

        System.out.println("🎉 恭喜！Java 物理引擎全线通关！");
    }
}

File: notebooks/16-bit_flip_max_consecutive_ones.py
class Solution:
    def reverseBits(self, num: int) -> int:
        num = num & 0xffffffff
        if num == 0xffffffff: return 32
        current,flipped, max_len = 0, 0, 0
        
        for _ in range(32):
            if (num & 1) == 1:
                current += 1
                flipped += 1
            else:
                flipped = current + 1
                current = 0           
            max_len = max(max_len, flipped)
            num >>= 1
        return max_len

if __name__ == "__main__":
    sol = Solution()
    
    print("🚦 开始执行位运算算法硬核全测...")
    
    # -------------------------------------------------------------
    # Case 1: 题目经典正数案例（1775 -> 二进制 ...11011101111）
    # 物理焊接后：3 + 1 + 4 = 8
    assert sol.reverseBits(1775) == 8, "测试失败：1775 应该输出 8"
    
    # Case 2: 题目经典正数案例（7 -> 二进制 ...0111）
    # 物理扩建后：3 + 1 = 4
    assert sol.reverseBits(7) == 4, "测试失败：7 应该输出 4"
    
    # -------------------------------------------------------------
    # Case 3: 极端边界哨兵（-1 -> 32位全1）
    # 物理天花板：直接触发 Guard Clause 吐出 32
    assert sol.reverseBits(-1) == 32, "测试失败：-1 应该输出 32"
    
    # Case 4: 闷骚负数断点（-2 -> 31个1 + 1个0）
    # 物理焊接后：31 + 1 = 32
    assert sol.reverseBits(-2) == 32, "测试失败：-2 应该输出 32"
    
    # Case 5: 刚才发生大震荡的负数（-3 -> 30个1 + 1个0 + 1个1）
    # 物理焊接后：30 + 1 + 1 = 31（座位被挤占，大部队缩水）
    assert sol.reverseBits(-3) == 32, "测试失败：-3 应该输出 32"
    
    # -------------------------------------------------------------
    # Case 6: 纯净零点（0 -> 32位全0）
    # 物理特权：随便把哪个 0 涂成 1，最长也只能拼出长度为 1 的铁轨
    assert sol.reverseBits(0) == 1, "测试失败：0 应该输出 1"
    
    print("✅ 恭喜，所有断言物理通关！代码质量稳如磐石。")